src/core/Global.ttslua

```diff
@@ -283,6 +284,134 @@ function getTokenIndex(token)
   end
 end
 
+function activeRedrawEffect(originParams)
+  dataFromLastReplacementEffect = originParams
+end
+
+function makeButtonsToRedraw()
+  if isTokenXMLActive == true then
+    broadcastToAll("Clear already active buttons first, then try again", "Red")
+    return
+  end
+
+  local tokensInPlay = getChaosTokensinPlay()
```

Comment:
This should use the existing `chaosTokens` table instead of the function that's only needed for outside access.